### ISSUE TITLE
@variant with selector argument fails to parse even with tailwindDirectives: true

### ISSUE THREAD

**Issue by eenlars** [biomejs/biome#8996]
## Environment

- Biome: 2.3.13 (also reproduced on 2.3.8)
- Tailwind CSS: v4
- OS: Linux

## Description

`@variant` with a selector argument (e.g. `@variant dark (.dark &);`) fails to parse even with `tailwindDirectives: true` enabled. The parser expects `{` but encounters `(`.

Notably, `@custom-variant` with the exact same selector syntax parses correctly. Only `@variant` with the selector form is broken.

Related: #8269 (which covers `@variant` inside `@utility` with numeric identifiers like `2xl`)

## Reproduction

**`biome.json`:**
```json
{
  "$schema": "https://biomejs.dev/schemas/2.3.13/schema.json",
  "css": {
    "parser": {
      "tailwindDirectives": true
    },
    "linter": { "enabled": true },
    "formatter": { "enabled": true }
  }
}
```

**`styles.css`:**
```css
@import "tailwindcss";

@variant dark (.dark &);

body {
  color: red;
}
```

**Command:** `biome check styles.css`

**Result:** 9 errors, starting with:

```
styles.css:3:15 parse ━━━━━━━━━━━━━━━━

  × expected `{` but instead found `(`

    1 │ @import "tailwindcss";
    2 │
  > 3 │ @variant dark (.dark &);
      │               ^

  i Remove (
```

The parse error cascades, breaking the rest of the file.

## Comparison: `@custom-variant` works fine

The same selector argument works correctly with `@custom-variant`:

```css
@import "tailwindcss";

@custom-variant dark (.dark &);

body {
  color: red;
}
```

This parses without errors — only a formatting diff is reported (spaces vs tabs).

## Tailwind CSS v4 reference

Per the [Tailwind CSS v4 directives documentation](https://tailwindcss.com/docs/functions-and-directives):

- **`@variant`** — Apply a Tailwind variant to styles in CSS. [Docs](https://tailwindcss.com/docs/functions-and-directives#variant-directive)
- **`@custom-variant`** — Define a new custom variant. [Docs](https://tailwindcss.com/docs/functions-and-directives#custom-variant-directive)

The `@variant` directive with a selector argument (`@variant dark (.dark &)`) was used in earlier Tailwind v4 betas. The stable v4 release uses `@custom-variant` for defining variants at the top level, while `@variant` is used inside rules to apply existing variants. Both forms should be parseable by Biome's Tailwind support.

## Expected behavior

Biome should parse `@variant dark (.dark &);` without errors when `tailwindDirectives: true` is enabled, matching how it already handles `@custom-variant dark (.dark &);`.

**Comment by dyc3** [comment]
It appears that this is legacy syntax. https://github.com/tailwindlabs/tailwindcss/blob/a8c54acaba7d51875bd332163754fb8bc40939a4/packages/tailwindcss/src/index.ts#L222

Given that its legacy syntax introduced by tailwind v4 **betas**, and that v4 has been out for some time now, I don't think we should expand our parser to support the syntax.
